spring-data-cassandra/src/main/java/org/springframework/data/cassandra/repository/support/SimpleCassandraRepository.java

```diff
@@ -247,6 +248,14 @@ public void deleteAll(Iterable<? extends T> entities) {
 		entities.forEach(this.operations::delete);
 	}
 
+	@Override
+	public void deleteAllById(Iterable<? extends ID> ids) {
+
+		Assert.notNull(ids, "The given Iterable of ids must not be null");
+
+		ids.forEach(id -> operations.deleteById(id, entityInformation.getJavaType()));
```

Comment:
We should use the same semantics as with `findAllById(…)` regarding the query mechanism to avoid individual database queries for each ID.